qbraid/runtime/oqc/device.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+"""
+Device class for OQC devices.
+
+"""
+from qcaas_client.client import QPUTask
+from scc.compiler.config import (
+    CompilerConfig,
+    MetricsType,
+    QuantumResultsFormat,
+    Tket,
+    TketOptimizations,
+)
+
+from qbraid.runtime.device import QuantumDevice
+from qbraid.runtime.enums import DeviceStatus, DeviceType
+from qbraid.transforms.qasm2.passes import rename_qasm_registers
+
+from .job import OQCJob
+
+results_format = {
+    "binary": QuantumResultsFormat().binary_count(),
+    "raw": QuantumResultsFormat().raw(),
+}
+
+metrics_type = {
+    "default": MetricsType.Default,
+    "empty": MetricsType.Empty,
+    "optimized_circuit": MetricsType.OptimizedCircuit,
+    "optimized_instruction_count": MetricsType.OptimizedInstructionCount,
+}
+
+optimizations = {
+    "clifford_simplify": Tket(TketOptimizations.CliffordSimp),
+    "context_simplify": Tket(TketOptimizations.ContextSimp),
+    "decompose_controlled_gates": Tket(TketOptimizations.DecomposeArbitrarilyControlledGates),
+    "default_mapping_pass": Tket(TketOptimizations.DefaultMappingPass),
+    "empty": Tket(TketOptimizations.Empty),
+    "full_peephole_optimise": Tket(TketOptimizations.FullPeepholeOptimise),
+    "globalise_phased_x": Tket(TketOptimizations.GlobalisePhasedX),
+    "kak_decomposition": Tket(TketOptimizations.KAKDecomposition),
+    "one": Tket(TketOptimizations.One),
+    "peephole_optimise_2q": Tket(TketOptimizations.PeepholeOptimise2Q),
+    "remove_barriers": Tket(TketOptimizations.RemoveBarriers),
+    "remove_discarded": Tket(TketOptimizations.RemoveDiscarded),
+    "remove_redundancies": Tket(TketOptimizations.RemoveRedundancies),
+    "simplify_measured": Tket(TketOptimizations.SimplifyMeasured),
+    "three_qubit_squash": Tket(TketOptimizations.ThreeQubitSquash),
+    "two": Tket(TketOptimizations.Two),
+}
+
+
+class OQCDevice(QuantumDevice):
+    """Device class for OQC devices."""
+
+    def __init__(
+        self, profile: "qbraid.runtime.TargetProfile", client: "qcaas_client.client.OQCClient"
+    ):
+        super().__init__(profile=profile)
+        self._client = client
+
+    @property
+    def client(self) -> "qcaas_client.client.OQCClient":
+        """Returns the client for the device."""
+        return self._client
+
+    def status(self) -> DeviceStatus:
+        if self.profile.get("device_type") == DeviceType.SIMULATOR:
+            return DeviceStatus.ONLINE
+        raise NotImplementedError("Only OQC simulators are supported")
```

Comment:
Can now use this code to get the actual status

```python
>>> from qcaas_client.client import OQCClient
>>> client = OQCClient(url="https://cloud.oqc.app/", authentication_token="...")
>>> devices = client.get_qpus()
>>> print(devices)
[{'active': True,
  'generation': 2,
  'id': 'qpu:uk:2:d865b5a184',
  'name': 'Lucy Simulator',
  'region': 'uk',
  'url': 'https://uk.cloud.oqc.app/d865b5a184'}]
```

We can just keep it simple and say that if `active` is `True` then device is online else offline